infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -865,6 +942,50 @@ private Map<UUID, StreamAddressSpace> getStreamsAddressesMap(
         return requestedAddressSpaces;
     }
 
+    /**
+     * Get a list of inactive streams. A stream is inactive if it's not a real Object stream or a checkpoint stream.
+     *
+     * @return a list of all inactive streams
+     */
+    private Set<UUID> getInactiveStreams(Set<UUID> existingStreams) {
+        if (existingStreams.isEmpty()) {
+            return Collections.emptySet();
+        }
+
+        CorfuRuntime corfuRuntime = getNewCorfuRuntime();
+
+        List<UUID> realStreams = corfuRuntime.getTableRegistry().listTables().stream()
+            .map(tableName -> CorfuRuntime.getStreamID(TableRegistry.getFullyQualifiedTableName(
+                tableName.getNamespace(), tableName.getTableName())))
+            .collect(Collectors.toList());
+        List<UUID> checkpointStreams = realStreams.stream()
+            .map(CorfuRuntime::getCheckpointStreamIdFromId)
+            .collect(Collectors.toList());
+
+        Set<UUID> inactiveStreams = existingStreams.stream()
+            .filter(stream -> !realStreams.contains(stream) && !checkpointStreams.contains(stream))
+            .collect(Collectors.toSet());
+
+        corfuRuntime.shutdown();
+        return inactiveStreams;
+    }
+
+    private CorfuRuntime getNewCorfuRuntime() {
+        final CorfuRuntime.CorfuRuntimeParameters params =
+                serverContext.getManagementRuntimeParameters();
+        params.setSystemDownHandlerTriggerLimit(SYSTEM_DOWN_HANDLER_TRIGGER_LIMIT);
+        final CorfuRuntime runtime = CorfuRuntime.fromParameters(params);
+        final Layout managementLayout = serverContext.copyManagementLayout();
+        // Runtime can be set up either using the layout or the bootstrapEndpoint address.
+        if (managementLayout != null) {
```

Comment:
This code is actually taken from ManagementServer.

Reply:
I suggest changing the if-condition to an assertion/precondition `if (managementLayout != null)`

The code **expects** the management layout to exist, otherwise the runtime doesn't know which servers to connect to.